control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go

```diff
@@ -185,6 +185,9 @@ func ReconcileAutoscalerDeployment(deployment *appsv1.Deployment, hcp *hyperv1.H
 		},
 		SetDefaultSecurityContext: setDefaultSecurityContext,
 	}
+	if hcp.Annotations[hyperv1.ControlPlanePriorityClass] != "" {
```

Comment:
Have we considered this be an API instead of an annotation?

I'm fine with it being an annotation but it'd be good to start discussing the path forward, e.g. something like
```
type Priority struct {
        // When set, overrides default and auto generated priorityClasses.
	// +optional
	PriorityClasses *PriorityClasses

	Auto bool
}

type PriorityClasses struct {
	ControlPlane string
	ApiCritial   string
	Etcd         string
}
```